service/service.go

```diff
@@ -151,7 +148,7 @@ func (srv *Service) Start(ctx context.Context) error {
 	)
 
 	// enable status reporting
-	srv.statusInit()
+	srv.telemetrySettings.Status.Ready()
```

Comment:
do we need to worry about status ever being `nil` in any of the places where its called without a check for nil first?

Reply:
This is intialized in `service.New(...)` and is in the same situation as `telemetrySettings.TracerProvider`, `telemetrySettings.MeterProvider` etc. The only way it would not be initialized would be skip the constructor, which I don't we need to be worried about.